lumigator/backend/backend/repositories/secrets.py

```diff
@@ -10,6 +10,18 @@ class SecretRepository(BaseRepository[SecretRecord]):
     def __init__(self, session: Session):
         super().__init__(SecretRecord, session)
 
+    def delete_secret(self, name: str) -> bool:
+        """Delete a secret identified by its name.
+
+        :param name: The name of the secret to be deleted
+        :returns: ``True`` if the secret was deleted, ``False`` if the secret was not found.
+        """
+        secret = self.get_secret_by_name(name)
+        if not secret:
+            return False
+
+        return bool(self.delete(secret.id))
```

Comment:
nit: is the query to get the secret before deleting it needed?

Might be a better option to just expect the caller to pass the secret id instead of its name, and do a delete operation directly instead of 2 sequential db operations